nextcord/application_command.py

```diff
@@ -2968,3 +3105,85 @@ def get_roles_from_interaction(state: ConnectionState, interaction: Interaction)
             ret.append(role)
 
     return ret
+
+
+def unpack_annotated(given_annotation, resolve_list: list = []) -> type:
+    origin = typing.get_origin(given_annotation)
+    if origin is Annotated:
+        located_annotation = MISSING
+        arg_list = typing.get_args(given_annotation)
+        # for i in range(1, len(arg_list)):
+        #     if arg_list[i] in resolve_list:
+        #         located_annotation = arg_list[i]
+        #         break
+        for arg in arg_list[1:]:
+            if arg in resolve_list:
+                located_annotation = arg
+                break
+
+        if located_annotation is MISSING:
+            located_annotation = arg_list[-1]
+
+        return located_annotation
+    else:
+        return given_annotation
+
+
+def unpack_annotation(
+    given_annotation: Any, annotated_list: List[type] = []
+) -> Tuple[List[type], list]:
+    """Unpacks the given parameter annotation into its components.
+
+    Parameters
+    ----------
+    given_annotation: Any
+        Given annotation to unpack. Should be from ``parameter.annotation``
+    annotated_list: List[:class:`type`]
+        List that the ``Annotated`` annotation should attempt to resolve to, from the 2nd arg to the right.
+
+    Returns
+    -------
+    Tuple[List[:class:`type`], :class:`list`]
+        A list of unpacked type annotations,
+        and a list of unpacked literal arguments.
+
+    """
+    type_ret = []
+    literal_ret = []
+    origin = typing.get_origin(given_annotation)
+    if origin is None:
+        # It doesn't have a fancy origin, just a normal type/object.
+        if isinstance(given_annotation, type):
+            type_ret.append(given_annotation)
+        else:
+            # If it's not a type and the origin is None, it's probably a literal.
+            literal_ret.append(given_annotation)
+
+    elif origin is Annotated:
+        # We want to treat it like the last annotation given.
+        # located_annotation = MISSING
+        # arg_list = typing.get_args(given_annotation)
+        # for i in range(1, len(arg_list)):
+        #     if arg_list[i] in annotated_list:
+        #         located_annotation = arg_list[i]
+        #         break
+        # for arg in arg_list[1:]:
+        #     if arg in annotated_list:
+        #
+        # if located_annotation is MISSING:
+        #     located_annotation = arg_list[-1]
```

Comment:
Shouldn't this commented code block also be removed?

Reply:
Ah, the leftovers of BSing your way through a feature and wanting to reference the old code, but forgetting to actually remove it. Removed.